spec/elastic_apm/trace_context/tracestate_spec.rb

```diff
@@ -22,31 +22,69 @@
 module ElasticAPM
   RSpec.describe TraceContext::Tracestate do
     describe '.parse' do
-      it 'parses a header' do
-        result = described_class.parse('a=b')
-        expect(result.values).to eq(['a=b'])
-      end
+      subject { described_class.parse(header) }
+
+      context 'without an es section' do
+        let(:header) { "a=b,c=d\ne=f" }
 
-      it 'handles multiple values' do
-        result = described_class.parse("a=b\nc=d")
-        expect(result.values).to eq(['a=b', 'c=d'])
+        it 'splits into individual entries by key' do
+          expect(subject.entries.keys).to eq %w[a c e]
+          expect(subject.entries.count).to be 3
+        end
       end
     end
 
     describe '#to_header' do
-      context 'with a single value' do
-        subject { described_class.parse('a=b') }
-        its(:to_header) { is_expected.to eq 'a=b' }
+      subject { described_class.parse(header) }
+
+      context 'with multiple sections' do
+        let(:header) { "a=b,c=d\ne=f" }
+        its(:to_header) { is_expected.to eq 'a=b,c=d,e=f' }
       end
+    end
+
+    context 'with an es field' do
+      subject { described_class.parse(header) }
 
-      context 'with multiple values' do
-        subject { described_class.parse("a=b\nc=d") }
-        its(:to_header) { is_expected.to eq 'a=b,c=d' }
+      let(:header) { "es=s:1;b:2,othervendor=na" }
+
+      it 'parses es field' do
+        expect(subject.entries['es'].value).to eq('s:1.0')
+        expect(subject.entries['othervendor'].value).to eq 'na'
       end
 
-      context 'with mixed' do
-        subject { described_class.parse("a=b,c=d\ne=f") }
-        its(:to_header) { is_expected.to eq 'a=b,c=d,e=f' }
+      context 'with bad values' do
+        [
+          ['es=xyz', nil],
+          ['es=s:foo', nil],
+          ['es=s:1.5', nil]
```

Comment:
@axw Do you feel like I should log these failures to parse?

Reply:
I wouldn't, or at least if you do make it trace level. Otherwise agent logs could be flooded by sending a lot of bogus requests.